CppExport/src/dependency_analysis/DependencyAnalyzer.cpp

```diff
@@ -0,0 +1,161 @@
+/***********************************************************************************************************************
+ **
+ ** Copyright (c) 2011, 2015 ETH Zurich
+ ** All rights reserved.
+ **
+ ** Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+ ** following conditions are met:
+ **
+ **    * Redistributions of source code must retain the above copyright notice, this list of conditions and the
+ **      following disclaimer.
+ **    * Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+ **      following disclaimer in the documentation and/or other materials provided with the distribution.
+ **    * Neither the name of the ETH Zurich nor the names of its contributors may be used to endorse or promote products
+ **      derived from this software without specific prior written permission.
+ **
+ **
+ ** THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ ** INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ ** DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ ** SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ ** SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ ** WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ ** OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **
+ **********************************************************************************************************************/
+
+#include "DependencyAnalyzer.h"
+
+#include "File.h"
+
+#include "OOModel/src/allOOModelNodes.h"
+
+namespace CppExport {
+
+void DependencyAnalyzer::associateNodeWithFile(Model::Node* node, QString name)
+{
+	// insert custom tree to file adjustments here
+	if (name == "Model/LoadedNode") name = "Model/PersistentStore";
+
+	auto it = nameToFileMap_.find(name);
+	if (it != nameToFileMap_.end())
+	{
+		(*it)->addNode(node);
+		return;
+	}
+
+	auto file = new File(name, node, new File(name, node));
+	nameToFileMap_.insert(name, file);
+	nodeToFileMap_.insert(node, file);
+}
+
+void DependencyAnalyzer::buildNodeToFileMap(Model::Node* name, QString namespaceName)
```

Comment:
This method name sounds very similar to the one above. Can we differentiate them more?